### ISSUE TITLE
About the model checkpoints

### ISSUE THREAD

**Issue by YukiFan** [wj-on-un/AffordanceLLM_implementation#1]
Hello, thank you very much for the code implementation of the paper "AffordanceLLM". May I ask if it is possible to open source the model checkpoints you pretrained?

**Comment by Alex-Beh** [comment]
same question, do you have the checkpoints?

**Comment by wj-on-un** [comment]
Hello, @YukiFan  @Alex-Beh 

I trained the model using the implementation code, but the affordance grounding result was strange, so I did not save the checkpoint separately.

**Comment by Alex-Beh** [comment]
Do you mind to share your training resources details?

Do you train on the entire AGD20k dataset?

And where to get this file [blip_laion_cc_sbu_558k.json](https://github.com/wj-on-un/AffordanceLLM_implementation/blob/d29bf16f8e67abb2aa7e0785f7312cb7e9895e47/train_aff_pretrain.sh#L5) ? I found [this](https://huggingface.co/datasets/liuhaotian/LLaVA-Pretrain/blob/main/blip_laion_cc_sbu_558k.json) but I am not sure whether this is the one you refer to.

**Comment by wj-on-un** [comment]
Exocentric images were not utilized because they do not have ground truth masks.

The dataset used consisted of egocentric images (maybe test set) with ground truth masks and images converted from depth, which were used as input. The affordance decoder was trained to generate ground truth masks.

In the AffordanceLLM paper it was mentioned:

> Implementation. We implement our model using PyTorch and HuggingFace. We initialize our model with LLama7B pretrained weights. Following LLaVA [40], we freeze the image encoder, pretrain the image projection layer to align OWL-ViT and LLama features, and then use GPT instructions to tune the language model. Finally, we add the mask encoder [29, 54] and tune the whole model on AGD20K [45], which has annotations of object affordance. We use eight NVIDIA A100 (40GB) to train our model, with Fully Sharded Data Parallel. We use a batch size of 4 and set the learning rate as 2e-5.

And the basic learning of LLaVA is done as follows. [[Link]](https://github.com/haotian-liu/LLaVA) - Train part

> LLaVA training consists of two stages: (1) feature alignment stage: use our 558K subset of the LAION-CC-SBU dataset to connect a frozen pretrained vision encoder to a frozen LLM; (2) visual instruction tuning stage: use 150K GPT-generated multimodal instruction-following data, plus around 515K VQA data from academic-oriented tasks, to teach the model to follow multimodal instructions.

First, we need to perform a pre-training process using the [laion-cc-sbu dataset](https://huggingface.co/datasets/liuhaotian/LLaVA-Pretrain) during the model feature alignment process.
Secon, we need to perform a GPT instruction tuning using [this](https://huggingface.co/datasets/liuhaotian/LLaVA-Instruct-150K/blob/main/README.md) (Maybe)

Then we need to learn the affordance decoder.

One thing I forgot to mention is that in my case I didn't train on the LLaVA-Instruct-150K dataset.
